openshift/catalog-sync/pkg/check/image.go

```diff
@@ -0,0 +1,88 @@
+package check
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/containers/image/v5/manifest"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/v2"
+)
+
+func AllImageChecks() []ImageCheck {
+	return []ImageCheck{
+		ImageIsManifest(),
+		ImageHasLabels(map[string]string{
+			"operators.operatorframework.io.index.configs.v1": "/configs",
+		}),
+	}
+}
+
+func ImageIsManifest() ImageCheck {
+	return ImageCheck{
+		Name: "IsImageManifest",
+		Fn: func(ctx context.Context, root ocispec.Descriptor, target oras.ReadOnlyTarget) error {
+			manifestReader, err := target.Fetch(ctx, root)
+			if err != nil {
+				return err
+			}
+			defer manifestReader.Close()
+			var imgManifest ocispec.Manifest
+			if err := json.NewDecoder(manifestReader).Decode(&imgManifest); err != nil {
+				return err
+			}
+			switch imgManifest.MediaType {
+			case ocispec.MediaTypeImageManifest, manifest.DockerV2Schema2MediaType:
+			default:
+				return fmt.Errorf("unrecognized manifest type: %s", imgManifest.MediaType)
+			}
+			return nil
+		},
+	}
+}
+
+func ImageHasLabels(expectedLabels map[string]string) ImageCheck {
+	return ImageCheck{
+		Name: "HasExpectedImageLabels",
```

Comment:
I noticed this in my code when I was looking at it the other day... Would it make sense to include a stringified version of `expectedLabels` in the `Name`? That way if this check is created multiple times with different label sets, the `Name` will ultimately still distinguish which sets of labels were not found.

Reply:
done